src/cc_catalog_airflow/dags/provider_api_scripts/flickr.py

```diff
@@ -258,6 +262,10 @@ def _process_image_data(image_data):
         logger.warning('No foreign_id in image_data!')
     foreign_landing_url = _build_foreign_landing_url(creator_url, foreign_id)
 
+    owner = image_data.get('owner').strip()
+    source = next((s for s in SUB_PROVIDERS if owner in SUB_PROVIDERS[s]),
+                  PROVIDER)
```

Comment:
Something I forgot in my earlier review:

Please pass `SUB_PROVIDERS` and `PROVIDER` in as parameters. It will make it easier to experiment with other sub-provider sets down the road, and makes testing more robust, since you can pass in precisely the subprovider list you want to test against.

Reply:
You mean pass them in as parameters to `_process_image_data`? Then we need to decide how far up the parameter passing should go. Should it be from where the `_process_interval` method is called from within the main method because that's the starting point of the flow?

Reply:
I was thinking of making them defaults in `_process_image_data.  That is, the signature would be:
```python
def _process_image_data(image_data, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
```
Then the further up functions don't need to know about them.  The point would be to enable passing different values for testing, and if someone wants to use the function in a not-yet-thought-of manner, but avoid having functions that are already using it needing more info than necessary to call the function.